### ISSUE TITLE
Multiple windows/local exploits causing NameError undefined local variable or method `sysinfo' since 6.3.47

### ISSUE THREAD

**Issue by ksb4ever** [rapid7/metasploit-framework#18647]
## Steps to reproduce

How'd you do it?

1. gain initial foothold (with meterpreter reverse_tcp as payload)
2. background session
3. use exploit/windows/local/tokenmagic
4. set session 1
5. run

Running on Kali Rolling updated.

## Expected behavior

> exploit 

[*] Started reverse TCP handler on x.x.x.x:4444 
[*] UAC is Enabled, checking level...
[+] Part of Administrators group! Continuing...
[+] UAC is set to Default
[+] BypassUAC can bypass this setting, continuing...
[*] Configuring payload and stager registry keys ...
[*] Executing payload: C:\Windows\SysWOW64\eventvwr.exe
[+] eventvwr.exe executed successfully, waiting 10 seconds for the payload to execute.
[*] Sending stage (175686 bytes) to y.y.y.y
[*] Meterpreter session 2 opened (x.x.x.x:4444 -> y.y.y.y:49231) at 2023-12-30 19:29:59 +0200
[*] Cleaning up registry keys ...

meterpreter > 

## Current behavior

[-] Exploit failed: NameError undefined local variable or method `sysinfo' for #<Module:exploit/windows/local/tokenmagic datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007faf88876538 @options={"WORKSPACE"=>#<Msf::OptString:0x00007faf886cdd80 @name="WORKSPACE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[],
[..]
@aliases={"reverseconnectretries"=>"stagerretrycount"}, @defaults={"PAYLOAD"=>"windows/x64/meterpreter/reverse_tcp", "WfsDelay"=>900}, @user_defined={}, @_module=#<Module:exploit/windows/local/tokenmagic datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007faf88cdb8f8 ...>]>>]>>]>
Did you mean?  @sysinfo
[*] Exploit completed, but no session was created.

### Metasploit version

Framework: 6.3.47-dev
Console  : 6.3.47-dev

## Additional Information
Debug file attached
[metasploit-local-debug.txt](https://github.com/rapid7/metasploit-framework/files/13798747/metasploit-local-debug.txt)

Works in previous versions including:
Framework: 6.3.46-dev
Console  : 6.3.46-dev

Modules found affected, list perhaps is not exhaustive:
exploit/windows/local/bypassuac_eventvwr
exploit/windows/local/ms13_053_schlamperei
exploit/windows/local/ms13_081_track_popup_menu
exploit/windows/local/ms14_058_track_popup_menu
exploit/windows/local/ms15_051_client_copy_image
exploit/windows/local/ntusermndragover
exploit/windows/local/ppr_flatten_rec
exploit/windows/local/tokenmagic

**Comment by nam3lum** [comment]
Wanted to highlight another module:
**multi/manage/autoroute**
Temporary workaround: comment out the informative messages and relaunch msfconsole.
Example: The fix below is for autoroute (/usr/share/metasploit-framework/modules/post/multi/manage/autoroute.rb:75):
```
print_status("Running module against target")# #{sysinfo['Computer']}")
```
![image](https://github.com/rapid7/metasploit-framework/assets/64528432/9e435f70-5208-41ee-b9b6-e5219766a5bf)

**Comment by AdaptOrDie247** [comment]
# Temporary Workaround

1. Backup the module file that you are trying to use. E.g.:
`mkdir ~/backups`
`cp /usr/share/metasploit-framework/modules/exploits/windows/local/ricoh_driver_privesc.rb ~/backups`

2. Open the affected module from its original location, using a text editor. E.g.:
`sudo mousepad /usr/share/metasploit-framework/modules/exploits/windows/local/ricoh_driver_privesc.rb`

3. Comment out the buggy code (in this case, it is just an architecture check, so as long as you are using the correct module for the target system, this shouldn't break anything). E.g.:
```
146    #if sysinfo['Architecture'] != payload.arch.first
147    # fail_with(Failure::BadConfig, 'The payload should use the same architecture as the target driver')
148    #end
```

4. Save the changes to the file.

5. Start (or restart) metasploit. Make sure you kill any sessions first. E.g.:
`sessions`
`sessions -k sessionid`
`exit`
`msfconsole -q`

6. Try running the module again. It may work, but if not, review and modify the module code as needed. Make sure you restart metasploit after each modification.

**Comment by doctorDAlvarez** [comment]
set Autocheck false worked for me. ;)

**Comment by DEFCESCO** [comment]
I've run into this same issue: 

Framework: 6.3.49-dev
Console  : 6.3.49-dev

```
msf6 > use multi/handler
[*] Using configured payload generic/shell_reverse_tcp
msf6 exploit(multi/handler) > 
msf6 exploit(multi/handler) > set payload windows/x64/meterpreter/reverse_tcp
payload => windows/x64/meterpreter/reverse_tcp
msf6 exploit(multi/handler) > set LHOST tun0
LHOST => tun0
msf6 exploit(multi/handler) > set LPORT 443
LPORT => 443
msf6 exploit(multi/handler) > set ExitOnSession false
ExitOnSession => false
msf6 exploit(multi/handler) > run -j
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.

[*] Started reverse TCP handler on 192.168.45.220:443 
msf6 exploit(multi/handler) > [*] Sending stage (200774 bytes) to 192.168.222.121
[*] Meterpreter session 1 opened (192.168.45.220:443 -> 192.168.222.121:59023) at 2024-01-03 12:24:10 -0500

msf6 exploit(multi/handler) > use multi/manage/autoroute
msf6 post(multi/manage/autoroute) > set session 1
session => 1
msf6 post(multi/manage/autoroute) > run

[-] Post failed: NameError undefined local variable or method `sysinfo' for #<Module:post/multi/manage/autoroute datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007f9274419458 @options={"WORKSPACE"=>#<Msf::OptString:0x00007f927458ff80 @name="WORKSPACE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Specify the workspace for this module", @default=nil, @enums=[], @owner=Msf::Module>, "VERBOSE"=>#<Msf::OptBool:0x00007f927458fbe8 @name="VERBOSE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Enable detailed status messages", @default=false, @enums=[], @owner=Msf::Module>, "SESSION"=>#<Msf::OptInt:0x00007f927458c3a8 @name="SESSION", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="The session to run this module on", @default=nil, @enums=[], @owner=Msf::Post>, "SUBNET"=>#<Msf::OptString:0x00007f9274593e78 @name="SUBNET", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Subnet (IPv4, for example, 10.10.10.0)", @default=nil, @enums=[], @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>, "NETMASK"=>#<Msf::OptString:0x00007f9274593c20 @name="NETMASK", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Netmask (IPv4 as \"255.255.255.0\" or CIDR as \"/24\"", @default="255.255.255.0", @enums=[], @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>, "CMD"=>#<Msf::OptEnum:0x00007f9274593630 @name="CMD", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc_string="Specify the autoroute command", @enums=["add", "autoadd", "print", "delete", "default"], @default="autoadd", @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>}, @aliases={}, @defaults={}, @user_defined={"SESSION"=>1, "VERBOSE"=>false, "NETMASK"=>"255.255.255.0", "CMD"=>"autoadd"}, @_module=#<Module:post/multi/manage/autoroute datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007f92745882d0 @options={"WORKSPACE"=>#<Msf::OptString:0x00007f927458ff80 @name="WORKSPACE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Specify the workspace for this module", @default=nil, @enums=[], @owner=Msf::Module>, "VERBOSE"=>#<Msf::OptBool:0x00007f927458fbe8 @name="VERBOSE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Enable detailed status messages", @default=false, @enums=[], @owner=Msf::Module>, "SESSION"=>#<Msf::OptInt:0x00007f927458c3a8 @name="SESSION", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="The session to run this module on", @default=nil, @enums=[], @owner=Msf::Post>, "SUBNET"=>#<Msf::OptString:0x00007f9274593e78 @name="SUBNET", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Subnet (IPv4, for example, 10.10.10.0)", @default=nil, @enums=[], @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>, "NETMASK"=>#<Msf::OptString:0x00007f9274593c20 @name="NETMASK", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Netmask (IPv4 as \"255.255.255.0\" or CIDR as \"/24\"", @default="255.255.255.0", @enums=[], @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>, "CMD"=>#<Msf::OptEnum:0x00007f9274593630 @name="CMD", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc_string="Specify the autoroute command", @enums=["add", "autoadd", "print", "delete", "default"], @default="autoadd", @owner=Msf::Modules::Post__Multi__Manage__Autoroute::MetasploitModule>}, @aliases={}, @defaults={}, @user_defined={"SESSION"=>1}, @_module=#<Module:post/multi/manage/autoroute datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007f92745882d0 ...>]>>]>>]>
Did you mean?  @sysinfo
[-] Call stack:
[-]   /usr/share/metasploit-framework/modules/post/multi/manage/autoroute.rb:75:in `run'
[*] Post module execution completed
msf6 post(multi/manage/autoroute) > options

Module options (post/multi/manage/autoroute):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   CMD      autoadd          yes       Specify the autoroute co
                                       mmand (Accepted: add, au
                                       toadd, print, delete, de
                                       fault)
   NETMASK  255.255.255.0    no        Netmask (IPv4 as "255.25
                                       5.255.0" or CIDR as "/24
                                       "
   SESSION  1                yes       The session to run this
                                       module on
   SUBNET                    no        Subnet (IPv4, for exampl
                                       e, 10.10.10.0)


View the full module info with the info, or info -d command.

msf6 post(multi/manage/autoroute) > version
Framework: 6.3.49-dev
Console  : 6.3.49-dev
```

**Comment by LongWayHomie** [comment]
Same issue here for _enum_logged_on_users_ post module. Also happened with UAC bypass module _bypassuac_windows_store_filesys_.

Framework: v6.3.47-dev

```
meterpreter > run post/windows/gather/enum_logged_on_users 

[-] Post failed: NameError undefined local variable or method `sysinfo' for #<Module:post/windows/gather/enum_logged_on_users datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007fcc60427e60 @options={"WORKSPACE"=>#<Msf::OptString:0x00007fcc51f6a7f0 @name="WORKSPACE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Specify the workspace for this module", @default=nil, @enums=[], @owner=Msf::Module>, "VERBOSE"=>#<Msf::OptBool:0x00007fcc51f6a4a8 @name="VERBOSE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Enable detailed status messages", @default=false, @enums=[], @owner=Msf::Module>, "SESSION"=>#<Msf::OptInt:0x00007fcc51f69990 @name="SESSION", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="The session to run this module on", @default=nil, @enums=[], @owner=Msf::Post>, "CURRENT"=>#<Msf::OptBool:0x00007fcc51f68dd8 @name="CURRENT", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="Enumerate currently logged on users", @default=true, @enums=[], @owner=Msf::Modules::Post__Windows__Gather__Enum_logged_on_users::MetasploitModule>, "RECENT"=>#<Msf::OptBool:0x00007fcc51f68b30 @name="RECENT", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="Enumerate recently logged on users", @default=true, @enums=[], @owner=Msf::Modules::Post__Windows__Gather__Enum_logged_on_users::MetasploitModule>}, @aliases={}, @defaults={}, @user_defined={"SESSION"=>3, "VERBOSE"=>false, "CURRENT"=>true, "RECENT"=>true}, @_module=#<Module:post/windows/gather/enum_logged_on_users datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007fcc5208eb18 @options={"WORKSPACE"=>#<Msf::OptString:0x00007fcc51f6a7f0 @name="WORKSPACE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Specify the workspace for this module", @default=nil, @enums=[], @owner=Msf::Module>, "VERBOSE"=>#<Msf::OptBool:0x00007fcc51f6a4a8 @name="VERBOSE", @advanced=true, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=false, @desc="Enable detailed status messages", @default=false, @enums=[], @owner=Msf::Module>, "SESSION"=>#<Msf::OptInt:0x00007fcc51f69990 @name="SESSION", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="The session to run this module on", @default=nil, @enums=[], @owner=Msf::Post>, "CURRENT"=>#<Msf::OptBool:0x00007fcc51f68dd8 @name="CURRENT", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="Enumerate currently logged on users", @default=true, @enums=[], @owner=Msf::Modules::Post__Windows__Gather__Enum_logged_on_users::MetasploitModule>, "RECENT"=>#<Msf::OptBool:0x00007fcc51f68b30 @name="RECENT", @advanced=false, @evasion=false, @aliases=[], @max_length=nil, @conditions=[], @fallbacks=[], @required=true, @desc="Enumerate recently logged on users", @default=true, @enums=[], @owner=Msf::Modules::Post__Windows__Gather__Enum_logged_on_users::MetasploitModule>}, @aliases={}, @defaults={}, @user_defined={}, @_module=#<Module:post/windows/gather/enum_logged_on_users datastore=[#<Msf::ModuleDataStoreWithFallbacks:0x00007fcc5208eb18 ...>]>>]>>]>
Did you mean?  @sysinfo
[-] Call stack:
[-]   /usr/share/metasploit-framework/modules/post/windows/gather/enum_logged_on_users.rb:110:in `run'
meterpreter > 
```

**Comment by smcintyre-r7** [comment]
I was able to reproduce this issue and get a fix proposed for it in #18667. Thanks everyone for raising this to our attention.
